reporting-anything/src/download_helpers.js

```diff
@@ -0,0 +1,214 @@
+    }
+
+      // force wait for any resize to load after the above DOM modification
+      await page.waitFor(1000);
+      switch (reportSource) {
+        case 'Dashboard':
+          await page.waitForSelector(SELECTOR.DASHBOARD, {
+            visible: true,
+          });
+          break;
+        case 'Visualization':
+          await page.waitForSelector(SELECTOR.VISUALIZATION, {
+            visible: true,
+          });
+          break;
+        case 'Notebook':
+          await page.waitForSelector(SELECTOR.NOTEBOOK, {
+            visible: true,
+          });
+          break;
+        default:
+          break;
+      }
+
+      await waitForDynamicContent(page);
+
+      let buffer;
+      // create pdf or png accordingly
+      if (format === FORMAT.PDF) {
+        const scrollHeight = await page.evaluate(
+          /* istanbul ignore next */
+          () => document.documentElement.scrollHeight
+        );
+    
+        buffer = await page.pdf({
+          margin: undefined,
+          width: 1680,
+          height: scrollHeight + 'px',
+          printBackground: true,
+          pageRanges: '1',
+        });
+      } else if (format === FORMAT.PNG) {
+        buffer = await page.screenshot({
+          fullPage: true,
+        });
+      }
+    
+      const fileName = `reporting_anything.${format}`;
+      const curTime = new Date();
+      const timeCreated = curTime.valueOf();
+      await browser.close();
+      const data = { timeCreated, dataUrl: buffer.toString('base64'), fileName };
+      await readStreamToFile(data.dataUrl, fileName);
+  } catch (e) {
+    console.log('error is', e);
+    process.exit(1);
+  }
+}
+
+const waitForDynamicContent = async (
+  page,
+  timeout = 30000,
+  interval = 1000,
+  checks = 5
+) => {
+  const maxChecks = timeout / interval;
+  let passedChecks = 0;
+  let previousLength = 0;
+
+  let i = 0;
+  while (i++ <= maxChecks) {
+    let pageContent = await page.content();
+    let currentLength = pageContent.length;
+
+    previousLength === 0 || previousLength != currentLength
+      ? (passedChecks = 0)
+      : passedChecks++;
+    if (passedChecks >= checks) {
+      break;
+    }
+
+    previousLength = currentLength;
+    await page.waitFor(interval);
+  }
+};
+
+const getReportSourceFromURL = (url) => {
+  if (url.includes('dashboards')) {
+    return 'Dashboard';
+  }
+  else if (url.includes('visualize')) {
+    return 'Visualization';
+  }
+  else if (url.includes('discover')) {
+    return 'Saved search';
+  }
+  else if (url.includes('notebooks')) {
+    return 'Notebook';
+  }
+  return 'Other';
+}
+
+export const readStreamToFile = async (
+  stream,
+  fileName
+) => {
+  let base64Image = stream.split(';base64,').pop();
+  fs.writeFile(fileName, base64Image, {encoding: 'base64'}, function (err) {
+    console.log('Downloaded report');
+  })
+};
+
+export const getFileFormatPrefix = (fileFormat) => {
+  var fileFormatPrefix = 'data:' + fileFormat + ';base64,';
```

Comment:
const instead of var?

Reply:
good catch, forgot to change that when adding the code